src/detect-engine-frame.c

```diff
@@ -473,8 +476,12 @@ static int FrameStreamDataInspectFunc(
 #endif
     BUG_ON(fsd->frame->len > 0 && (int64_t)data_len > fsd->frame->len);
 
+    if (data_offset > UINT32_MAX) {
+        SCLogDebug("DETECT_ENGINE_INSPECT_SIG_NO_MATCH data_offset > UINT32_MAX");
+        return more_chunks;
+    }
     const bool match = DetectEngineContentInspection(det_ctx->de_ctx, det_ctx, s, engine->smd, p,
```

Comment:
What do you think about `DetectEngineContentInspection` taking a u32 as input but every caller seems to pass a u64 ?

Reply:
the size of the data passed should never even get close to the limit of a u32. But we do work with offsets that are higher, like in streaming buffers.

Reply:
So it must be broken currently as `DetectEngineContentInspection` silently casts the u64 to u32, am I getting it right ?